### PR TITLE
Move commit-specific methods from ZCL_ABAPGIT_BRANCH_OVERVIEW to new class ZCL_ABAPGIT_GIT_COMMIT

### PR DESCRIPTION
**Pull request by mariusraht2**
Related to #3040

It must be possible for e.g. feature #3040 to get commits from remote. We have already logic implemented that does exactly that in `ZCL_ABAPGIT_BRANCH_OVERVIEW`. Therefore I want to reuse it but the visibility and type of methods needed aren't allowing exactly that. One would have to create a BRANCH_OVERVIEW object which isn't needed in certain scenarios. Additionally the constructor would execute a lot of unnecessary functions. Plus we should move those methods to object-specific classes instead of hiding them in classes only related to one specific overview UI/functionality.

**Example:**
```
zcl_abapgit_git_commit=>get_by_commit( io_repo         = io_repo
				       iv_commit_hash  = io_repo->get_branch_hash_remote( )
                                       iv_deepen_level = 99 )
```

**Change Log:**
- Created new class `ZCL_ABAPGIT_GIT_COMMIT`
- Moved commit-specific methods from `ZCL_ABAPGIT_BRANCH_OVERVIEW` to `ZCL_ABAPGIT_GIT_COMMIT` (functionality didn't change for copied/existing methods)
- Changed methods visibility (from private to public) and type (from instance to class method)
- Added method `get_by_commit` that gets the commits from remote and sorts them by their hierarchy to eachother
- Added method `get_by_branch` which is the equivalent of `get_by_commit` for the branch name as import parameter
- Moved test methods/class to `ZCL_ABAPGIT_GIT_COMMIT` because all tests within `ZCL_ABAPGIT_BRANCH_OVERVIEW` are commit-specific

### CODE REVIEW

File: src/git/zcl_abapgit_git_commit.clas.abap [hunk 1]
```diff
@@ -0,0 +1,233 @@
+"! <p class="shorttext synchronized" lang="en">Git Commit</p>
+CLASS zcl_abapgit_git_commit DEFINITION
+  PUBLIC
+  CREATE PUBLIC .
+
+  PUBLIC SECTION.
+    CLASS-METHODS get_from_remote
+      IMPORTING
+        !io_repo          TYPE REF TO zcl_abapgit_repo_online
```
Comment (larshp) [inline src/git/zcl_abapgit_git_commit.clas.abap:9]: Hi,
I'm trying to keep the repository concept outside of the `/git/` package, so it theoretically can be used for other things.
Suggest splitting this method into two, `get_by_branch` and `get_by_commit`, and then the URL and branch/sha1 can be passed

**Comment by mariusraht2** [comment]
@larshp Done :) Thanks for reviewing all my PRs with such patience 👍

File: src/git/zcl_abapgit_git_commit.clas.abap [hunk 2]
```diff
@@ -0,0 +1,257 @@
+"! <p class="shorttext synchronized" lang="en">Git Commit</p>
+CLASS zcl_abapgit_git_commit DEFINITION
+  PUBLIC
+  CREATE PUBLIC .
+
+  PUBLIC SECTION.
+    CLASS-METHODS get_by_branch
+      IMPORTING
+        !iv_branch_name   TYPE string
+        !iv_repo_name     TYPE string
```
Comment (larshp) [inline src/git/zcl_abapgit_git_commit.clas.abap:10]: seen with git eyes, a repository name does not exist, its an URL endpoint containing git objects

in the code, `iv_repo_name` is only used for a progress indicator, suggest using the URL for the progress indicator instead

**Review by larshp — CHANGES_REQUESTED** [review]

**Review by larshp — APPROVED** [review]

**Comment by larshp** [comment]
thanks 👍

**Pull request merged**